Client/Frontend/Browser/TabToolbar.swift

```diff
@@ -46,15 +55,40 @@ open class TabToolbarHelper: NSObject {
 
     let ImageReload = UIImage.templateImageNamed("nav-refresh")
     let ImageStop = UIImage.templateImageNamed("nav-stop")
+    let ImageSearch = UIImage.templateImageNamed("search")
 
-    var loading: Bool = false {
-        didSet {
-            if loading {
+    func setMiddleButtonState(_ state: MiddleButtonState) {
```

Comment:
Could set access control of `setMiddleButtonState` method to `fileprivate setMiddleButtonState`